src/compare/__tests__/computeSsim.test.ts

```diff
@@ -0,0 +1,121 @@
+import { computeMse } from '..';
+import { Image, ImageColorModel } from '../..';
+import { computeSsim } from '../computeSsim';
+
+test('twice the same image', async () => {
+  const image = testUtils.createGreyImage([[5, 5, 5, 10, 10, 10, 15, 15, 15]]);
+  const other = image;
+  expect(computeSsim(image, other).mssim).toBe(1);
+});
+
+test('should be symetrical', async () => {
+  const image = testUtils.createGreyImage([[1, 2, 3, 4, 5]]);
+  const other = testUtils.createGreyImage([[0, 0, 0, 0, 0]]);
+  expect(computeSsim(image, other).mssim).toBe(computeSsim(other, image).mssim);
+});
+
+test('ssim should be zero', async () => {
+  const image = new Image(11, 11, { colorModel: ImageColorModel.GREY });
+  const other = Image.createFrom(image).fill(255);
+
+  expect(image === other).toBe(false);
+  expect(computeSsim(image, other).mssim).toBeCloseTo(0, 3);
+});
+
+test('ssim should be -1 (anti-correlated images)', async () => {
+  const image = testUtils.createGreyImage([
+    [0, 0, 0],
+    [255, 255, 255],
+  ]);
+  const other = testUtils.createGreyImage([
+    [255, 255, 255],
+    [0, 0, 0],
+  ]);
+  expect(image === other).toBe(false);
+  expect(computeSsim(image, other).mssim).toBeCloseTo(-1, 2);
+});
+
+test('original with itself', async () => {
+  const original = testUtils.load('ssim/ssim-original.png');
+  const other = original;
+  expect(computeSsim(original, other).mssim).toBe(1);
+  expect(computeMse(original, other)).toBe(0);
+});
+
+// In the following tests the SSIM expected values were computed with Matlab.
+
+test('more contrast', async () => {
+  let original = testUtils.load('ssim/ssim-original.png');
+  let other = testUtils.load('ssim/ssim-contrast.png');
+  expect(computeSsim(original, other).mssim).toBeCloseTo(0.8201, 1);
+});
```

Comment:
Did you find the problem you hade with the comparison to Matlab?

Reply:
Yes, I was passing GREY images, whereas the library only accepts RGBA images....